### PR TITLE
Tweaks for various test failures

### PR DESCRIPTION
**Pull request by kees**
These are some tweaks to improve test failure readability I ran into with ttyname and netfilter.

### CODE REVIEW

**Comment by rocallahan** [comment]
Thank you very much!

File: CMakeLists.txt [hunk 1]
```diff
@@ -66,6 +66,12 @@ else()
 endif()
 configure_file(src/preload/rr_page.ld.in src/preload/rr_page.ld @ONLY)
 
+include(CheckCCompilerFlag)
+CHECK_C_COMPILER_FLAG("-fmacro-prefix-map=$(srctree)/=" SUPPORTS_MACRO_PREFIX_MAP)
+if (SUPPORTS_MACRO_PREFIX_MAP)
+  set(FLAGS_COMMON "${FLAGS_COMMON} -fmacro-prefix-map=${CMAKE_SOURCE_DIR}/=")
```
Comment (yuyichao) [inline CMakeLists.txt:72]: I think this should be added to the test flags instead and not the common flags. For the installed binary, the distribution package builder may set this to point to the actual installation location of the source files instead.

Comment (rocallahan) [reply CMakeLists.txt:72]: I don't understand, sorry. The goal is to make __FILE__ more readable. I think it makes sense to do this in distribution packages as well as local builds. What's the problem?

Comment (yuyichao) [reply CMakeLists.txt:72]: Distribution can set this to the actual installation path of the source (when the source code is installed as part of the debug package.

Comment (rocallahan) [reply CMakeLists.txt:72]: What do you mean by "this"? `CMAKE_SOURCE_DIR`?

Comment (yuyichao) [reply CMakeLists.txt:72]: fmacro-prefix-map

Comment (yuyichao) [reply CMakeLists.txt:72]: In particular, the value it maps the cmake source path into. Here the mapped value is just empty.

**Comment by rocallahan** [comment]
I don't think it's particularly useful for distributions to set this. It only appears in rr messages, it's not changing debuginfo as far as I know.

File: CMakeLists.txt (hunk 1 above)
Comment (rocallahan) [reply CMakeLists.txt:72]: I don't think it's particularly useful for distributions to set this. It only appears in rr messages, it's not changing debuginfo as far as I know.

Comment (rocallahan) [reply CMakeLists.txt:72]: Having said that, this change doesn't seem to work locally.

Comment (rocallahan) [reply CMakeLists.txt:72]: OK, it only works with the `make` backend, not `ninja`...

Comment (rocallahan) [reply CMakeLists.txt:72]: Fixed that with b9f4eea09d1038347fc009f8ce6c7752e0951c06

**Comment by yuyichao** [comment]
> I don't think it's particularly useful for distributions to set this.

Well, distributions do set that, at least ArchLinux does (in forms of the file-prefix-map, which covers this).

> It only appears in rr messages, it's not changing debuginfo as far as I know.

Correct, but unlike messages from the tests, which may happen more regularly and are for developers that already knows what they are looking at, the messages at runtime are for users and developers that may not be as familiar with RR internals. I do find seeing the full path to the source helpful for helping me finding the right source file before attaching a debugger/if I couldn't get it in a debugger.

**Comment by rocallahan** [comment]
On balance I think the change is a good one and I don't want to revert it, but if you submit a patch to let distro settings override this, I'd take it.

**Comment by kees** [comment]
As long as the option ordering puts the distro flags last, it'll have priority, so it's possible things already work correctly:
https://godbolt.org/z/1M3vGEWPq

**Comment by yuyichao** [comment]
> As long as the option ordering puts the distro flags last, it'll have priority

Yes, that's exactly my plan. What I need to check is what happens if the prefix specified are different.
Also I'm fairly sure that the distribution flags usually comes from the environment variable and they got picked up in `CMAKE_<C|CXX>_FLAGS`. These are actually in front so the flags you set here actually overwrite that, which is why I brought this up in the first place.

**Pull request merged**